doc/ideas/savefileformat.md

```diff
@@ -0,0 +1,446 @@
+| 490-491 | byte[3]   | reserved                          | must be 0 |
+| 492-516 | byte[25]  | timestamp                         | ISO 8601: 2017-01-11T22:07:12+00:00 |
+| 517-772 | byte[216] | crypto key                        | must be 0, if crypto is not used |
+| 773-788 | byte[16]  | crypto iv                         | must be 0, if crypto is not used |
+| 789-852 | byte[64]  | sha512 over bytes 000-788         | header checksum, used to check if header is corrupted |
+
+### 4 String Table
+
+This table stores all the strings needed in the savegame, so they can compressed easily and simply referenced by a hashmap. The order of appearance in this list is there referrence id.
+
+all bytes are offsets to "string table offset" (Header Section)
+
+| bytes        | type                                 | name            | info                                                  |
+| ------------ | ------------------------------------ | --------------- | ----------------------------------------------------- |
+| 0-3          | uint32                               | length          | length of this compressed segment in bytes            |
+| 4-7          | uint32                               | string count    | string count                                          |
+| 8-(length-1) | string[string count] list of strings | these strings will be referenced by others:  id=count                   |
+
+### 5 Key Value Table
+
+all bytes are offsets to "key value table offset" (Header Section)
+
+| bytes        | type   | name                 | info                                       |
+| ------------ | ------ | -------------------- | ------------------------------------------ |
+| 0-3          | uint32 | length               | length of this compressed segment in bytes |
+| 4-7          | uint32 | key/value pair count | attribute count                            |
+| 8-(length-1) | ..     | list of single value | Single Value @5.1, order = id              |
+
+#### 5.1 Single Value
+
+
+| bytes | type  | name             | info                                  |
+| ----- | ----- | ---------------- | ------------------------------------- |
+| 0-0   | uint8 | basic type       | one of the basic type below           |
+| 1-..  | uint+ | string reference | reference to String Table (Section 4) |
+
+type: 0=uint8, 1=uint16, 2=uint32, 3=uint64, 4=int8, 5=int16, 6=int32, 7=int64, 8=uint+, 9=int+, 10=f32, 11=f64, 12=string
+
+### 6 Game Information:
+
+gameinformation will be compressed: bytes 4 : length-1
+
+all bytes are offsets to gameinformation offset
+
+| bytes  | type        | name                     | info |
+| ------ | ----------- | ------------------------ | --- |
+| 0-3    | uint32      | length                   | length of this compressed segment in bytes |
+| 4-4    | int8        | speed                    | game speed relative to normal in 1/10 steps 90 => 9 times faster, -4 => 0.4 times normal speed |
```

Comment:
as idea, it's ok, but we really can't guess how we can save our simulation efficiently. also, we can even represent all the game information as keyframes of our curves, which then just remains the same all the game.

So the much simpler approach is to create a generic curve storage, which we can serialize and unserialize and snapshot.